Functional group addition

talagayev opened the pull request:
Changes made in this PR:

- Addition of functional groups
- coloring of functional groups atoms and bonds
- black formatting of the code

talagayev committed:
Update mdonatello.py

Adjustment for the aromatic pharmacophore to use the isAromatic instead

talagayev committed:
Update mdonatello.py

First implementation of functional groups

talagayev committed:
Update mdonatello.py

Added additional functional groups and made color adjustments according to the element present

talagayev committed:
Update mdonatello.py

Added bonds coloring

talagayev committed:
Update mdonatello.py

Added Black formatting

talagayev committed:
Update mdonatello.py

starting doing Type Checking&Hinting

talagayev committed:
Update mdonatello.py

Type Checking&Hinting

talagayev committed:
Update mdonatello.py

restructure of code for physiochem properties and hbond acceptors donors

talagayev committed:
Update mdonatello.py

moved pharmacophore and functional groups to separate classes

talagayev committed:
Update mdonatello.py

fixed functional group error

talagayev committed:
Update mdonatello.py

talagayev committed:
Update mdonatello.py

moved highlighting of molecules

talagayev committed:
Update mdonatello.py

moved highlight and drawer

talagayev committed:
Update mdonatello.py

fixed error and changed func group representation

talagayev committed:
Update mdonatello.py

changed functional groups to checkboxes again

talagayev committed:
Update mdonatello.py

fixed error for input

talagayev committed:
Update mdonatello.py

adjusted the functional group highlighting of the bonds

richardjgowers commented on mdonatello/mdonatello.py:
```diff
@@ -116,78 +159,229 @@ def draw_molecule(self, mol, show_atom_indices, width, height):
+        )
+ 
+    def calculate_functional_groups(self, mol):
+        functional_groups = {
```
there's a dictionary being created here, but it's never being used as a dictionary because you're only accessing it via `.items()`.  instead you could just create a list of tuples of (fg_name, fg_SMARTS).  so:
```python
functional_groups = [
    ('Hydroxyl_group (-OH)', '[OX2H]'),
    ...
]
```

Similarly I think the output of this function is a dict but never used as a dict but is instead immediately traversed via `.items()`.  If you're immediately consuming the output of this function it could even be a generator...

richardjgowers commented on mdonatello/mdonatello.py:
```diff
@@ -116,78 +159,229 @@ def draw_molecule(self, mol, show_atom_indices, width, height):
+        )  # Default to grey if not specified
+
+    def get_color_for_functional_group(self, fg):
+        parts = fg.split('(')
```
this is a place where a dict **could** be used, you could have a lookup of SMARTS patterns to desired colours, and it would be less prone to false colours etc

talagayev committed:
Update mdonatello.py

Added rotatable bonds

talagayev committed:
Update mdonatello.py

Adjusted the printing of the properties to have correct values

talagayev committed:
Update test_mdonatello.py

Added tests for property classes

talagayev committed:
Update __init__.py

talagayev committed:
Update mdonatello.py

Added partial charges and adjusted the code with black formatting

talagayev committed:
Update mdonatello.py

Added Murcko and Stereocenters and fixed Rotatable bonds recognition of atom indices

talagayev committed:
Moved functional group and pharmacophore classes to mapper.py

talagayev committed:
Separated the highlighting into classes and moved with MoleculeDrawer to drawer.py

talagayev committed:
Create properties.py

talagayev committed:
Adjusted mdonatello.py with the new code adjustment

talagayev committed:
Update mdonatello.py

talagayev committed:
Separated Moleculevisualizer in more classes, added documentation and removed not used imports

talagayev committed:
Create test_mapper.py

talagayev committed:
Create CI_CD.yml

talagayev committed:
Update CI_CD.yml

removed env update

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

Added codecov

talagayev committed:
Delete mdonatello/tests/test_mdonatello.py

currently removed to adjust new tests and test codecov

talagayev committed:
Update CI_CD.yml

adjusting code cov

talagayev committed:
Update CI_CD.yml

talagayev committed:
Create test_drawer.py

Adde first test in drawer

talagayev committed:
Update CI_CD.yml

talagayev committed:
Create test_mdonatello.py

talagayev committed:
Added test pdb file

talagayev committed:
Update test_mdonatello.py

talagayev committed:
Update test_mdonatello.py

talagayev committed:
Added new tests

talagayev committed:
Added Figure for Display.rst

talagayev committed:
Added Display in index.rst

talagayev committed:
Added display.rst

talagayev committed:
Update getting_started.rst

talagayev committed:
Update getting_started.rst

talagayev committed:
Added IpyWidgets Matrix

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Add files via upload

talagayev committed:
Update conf.py

talagayev committed:
Delete docs/source/_static/logo/Mdonatello_logo.png

talagayev committed:
Add files via upload

talagayev committed:
Update conf.py

talagayev committed:
Delete docs/source/_static/logo/Mdonatello_logo.png

talagayev committed:
Add files via upload

talagayev committed:
Delete docs/source/_static/logo/Mdonatello_logo.png

talagayev committed:
Add files via upload

talagayev committed:
Update getting_started.rst

talagayev committed:
Update getting_started.rst

talagayev committed:
Update README.md

talagayev committed:
Update README.md

talagayev committed:
Update getting_started.rst

talagayev committed:
Create CI_CD_MacOS.yml

talagayev committed:
Rename CI_CD.yml to Linux_CI_CD.yml

talagayev committed:
Rename Linux_CI_CD.yml to CI_CD.yml

talagayev committed:
Update and rename CI_CD.yml to Linux_CI_CD.yml

talagayev committed:
Update Linux_CI_CD.yml

talagayev committed:
Update and rename CI_CD_MacOS.yml to MacOS_CI_CD.yml

talagayev committed:
Create Windows_CI_CD.yml

talagayev committed:
Update Windows_CI_CD.yml

talagayev committed:
Update Windows_CI_CD.yml

talagayev committed:
Merge pull request #19 from talagayev/Class_code_movement

Class code movement

talagayev committed:
Merge pull request #18 from talagayev/Rotatable_bond_addition

Rotatable bond addition

talagayev committed:
Merge pull request #17 from talagayev/Code_restructure

Code class restructure
